core/deployment/src/main/java/io/quarkus/deployment/dev/devservices/GlobalDevServicesConfig.java

```diff
@@ -4,36 +4,37 @@
 import java.util.Optional;
 import java.util.function.BooleanSupplier;
 
-import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigRoot;
-import io.smallrye.config.ConfigMapping;
-import io.smallrye.config.WithDefault;
 
 /**
  * Dev Services
+ *
+ * @deprecated Please, use {@link DevServicesConfig} instead.
  */
-@ConfigMapping(prefix = "quarkus.devservices")
-@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
-public interface GlobalDevServicesConfig {
+@Deprecated(forRemoval = true)
+@ConfigRoot(name = "devservices")
+public class GlobalDevServicesConfig {
 
     /**
      * Global flag that can be used to disable all Dev Services. If this is set to false then Dev Services will not be used.
      */
-    @WithDefault("true")
-    boolean enabled();
+    @ConfigItem(defaultValue = "true", generateDocumentation = false)
```

Comment:
Could you add `@ConfigDocIgnore` for all these individual properties?

In the new model, I actually keep the deprecated stuff around as it could be useful for IDEs to know the properties is known. But in this case, we want to make sure only the new ones are around.

Reply:
Yes, I've tried, but `@ConfigDocIgnore` is for method only, I had a look into https://github.com/quarkusio/quarkus/blob/59bd3858ac730fe0d0691427e2b9082d4797dc25/core/processor/src/main/java/io/quarkus/annotation/processor/documentation/config/scanner/ConfigAnnotationScanner.java#L423-L452 and `generateDocumention` should serve the same purpose?

Reply:
Ah, yes, it should do the trick!